### ISSUE TITLE
How can i solve the problem with model.predict() ?

### ISSUE THREAD

**Issue by SamMohel** [keras-team/keras#14931]
i encountered a problem in predict .. i got same predict sentences every time with using model.predict () i searched more and found that i need to increase epochs. but how can i decide that this number of epochs will be good or what is epoch depend on ? 
my data training is 90000 and testing is 40000 .. i used one epoch only so i got the problem .  should i reduce learning rate also ?

**Comment by namaannn** [comment]
@SamMohel increase the learning rate as model takes time to learn the data.

**Comment by SamMohel** [comment]
thanks for replying .. can this be a solution but learning rate start from low value 
```
def step_decay(epoch):
  initial_rate = 2e-6
  factor = int(epoch / 20)
  lr = initial_rate * (10 ** factor)
  return lr

def define_model(vocab_size, max_length, curr_shape):
    inputs1 = Input(shape=curr_shape)
    fe1 = inputs1
    fe2 = Dense(256, activation='relu')(fe1)
    model = tf.keras.models.Sequential()
    inputs2 = Input(shape=(max_length,))
    se1 = Embedding(vocab_size, 256, mask_zero=True)(inputs2)
    se2 = Dropout(0.2)(se1)
    se3 = LSTM(256)(se2)
    decoder1 =Concatenate()([fe2, se3])
    decoder2 = Dense(256, activation='relu')(decoder1)
    outputs = Dense(vocab_size, activation='softmax')(decoder2)
    opt =tf.keras.optimizers.Adam(learning_rate= 2e-6)
    model.compile(loss='categorical_crossentropy', optimizer=opt, metrics=['accuracy'])
    model.summary()
    return model
lr_schedule =  LearningRateScheduler(step_decay)
callbacks=[checkpoint,lr_schedule]
history = model.fit(train_generator, epochs=100, steps_per_epoch=train_steps, verbose=1, callbacks=callbacks, validation_data=val_generator, validation_steps=val_steps,batch_size=32)
```

**Comment by jvishnuvardhan** [comment]
@SamMohel  I think this is more of a support issue that is good for [TF Forum](https://discuss.tensorflow.org/)
There are more [tutorials](https://www.tensorflow.org/text/tutorials/text_generation) that may help you. Thanks!

This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [TF Forum](https://discuss.tensorflow.org/). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!
